src/components/NavBar/MenuItemMobile/index.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Button,
+  Typography,
+  withStyles,
+  WithStyles,
+  createStyles,
+  Switch,
+  Grid,
+} from '@material-ui/core';
+import MuiAccordion from '@material-ui/core/Accordion';
+import MuiAccordionSummary from '@material-ui/core/AccordionSummary';
+import MuiAccordionDetails from '@material-ui/core/AccordionDetails';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
+
+import { LayerType, TableType } from '../../../config/types';
+import { addLayer, removeLayer } from '../../../context/mapStateSlice';
+import { loadTable } from '../../../context/tableStateSlice';
+import { layersSelector } from '../../../context/mapStateSlice/selectors';
+import { menuList } from '../utils';
+
+function MenuItemMobile({ classes }: MenuItemMobileProps) {
+  const [expanded, setExpanded] = useState<string>('');
+
+  const selectedLayers = useSelector(layersSelector);
+
+  const dispatch = useDispatch();
+
+  const handleChange = (panel: string) => (
+    event: object,
+    newExpanded: boolean,
+  ) => {
+    setExpanded(newExpanded ? panel : '');
+  };
+
+  const toggleLayerValue = (prevChecked: boolean, layer: LayerType) => {
+    if (prevChecked) {
+      dispatch(removeLayer(layer));
+    } else {
+      dispatch(addLayer(layer));
+    }
+  };
+
+  const showTableClicked = (table: TableType) => {
+    dispatch(loadTable(table.id));
+  };
+
+  const Accordion = withStyles({
+    root: {
+      '&:before': {
+        display: 'none',
+      },
+      '&$expanded': {
+        margin: 'auto',
+        marginLeft: 0,
+        width: '100%',
+      },
+      boxShadow: 'none',
+    },
+    expanded: {},
+  })(MuiAccordion);
+
+  const AccordionSummary = withStyles(theme => ({
```

Comment:
I think you could create classes to pass to the different elements instead and generate styles only once. For example, look at how it is implemented in https://github.com/WFP-VAM/prism-frontend/blob/master/src/components/MapView/Legends/index.tsx

Reply:
@ericboucher thanks for the suggestion. I have updated the codes in the new commit